src/fsharp/Optimizer.fs

```diff
@@ -3112,16 +3121,22 @@ and OptimizeModuleDef cenv (env,bindInfosColl) x =
     | TMDefRec(isRec,tycons,mbinds,m) -> 
         let env = if isRec then BindInternalValsToUnknown cenv (allValsOfModDef x) env else env
         let mbindInfos,(env,bindInfosColl) = OptimizeModuleBindings cenv (env,bindInfosColl) mbinds
-        let mbinds,minfos = List.unzip mbindInfos
-        let binds = minfos |> List.choose (function Choice1Of2 (x,_) -> Some x | _ -> None)
-        let binfos = minfos |> List.choose (function Choice1Of2 (_,x) -> Some x | _ -> None)
-        let minfos = minfos |> List.choose (function Choice2Of2 x -> Some x | _ -> None)
-
+        let mbinds = List<_>()
```

Comment:
I prefer the original immutable code here. TMDefRec is relatively rare so I'm certain this doesn't dominate CPU perf.

(BTW if we were to use a mutable list, then please use ``ResizeArray``)